pkg/controller/hpaenforcement_controller.go

```diff
@@ -0,0 +1,555 @@
+					APIVersion: workload.GetObjectKind().GroupVersionKind().GroupVersion().String(),
+					Kind:       workload.GetObjectKind().GroupVersionKind().Kind,
+				},
+				MinReplicaCount: &min,
+				MaxReplicaCount: &max,
+				Triggers:        scaleTriggers,
+			},
+		}
+		// Add triggers
+		if result, err := controllerutil.CreateOrUpdate(ctx, r.Client, &scaledObj, func() error {
+			scaledObj.Spec = kedaapi.ScaledObjectSpec{
+				ScaleTargetRef: &kedaapi.ScaleTarget{
+					Name:       workload.GetName(),
+					APIVersion: workload.GetObjectKind().GroupVersionKind().GroupVersion().String(),
+					Kind:       workload.GetObjectKind().GroupVersionKind().Kind,
+				},
+				MinReplicaCount: &min,
+				MaxReplicaCount: &max,
+				Triggers:        scaleTriggers,
+			}
+			return nil
+		}); err != nil {
+			logger.V(0).Error(err, "Error creating or updating scaledobject")
+			return ctrl.Result{}, err
+		} else {
+			logger.V(0).Info(fmt.Sprintf("Result of the create or update operation is '%s\n'", result))
+			soString, _ := json.MarshalIndent(scaledObj, "", "   ")
+			logger.V(0).Info(fmt.Sprintf("ScaledObject created. %s\n", soString))
+		}
+
+	} else {
+		logger.V(0).Info("Skipping creating ScaledObject for workload as the controller is deployed in dryRun mode.", "workload", workload.GetName())
+		return ctrl.Result{}, nil
+		//	 TODO: log metric
+	}
+
+	statusPatch, conditions = CreatePolicyPatch(policyreco, conditions, v1alpha1.HPAEnforced, metav1.ConditionTrue, HPAEnforcedReason, HPAEnforcedMessage)
+	if err := r.Status().Patch(ctx, statusPatch, client.Apply, getSubresourcePatchOptions(HPAEnforcementCtrlName)); err != nil {
+		logger.Error(err, "Error updating the status of the policy reco object")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	r.Recorder.Event(&policyreco, eventTypeNormal, "ScaledObjectCreated", fmt.Sprintf("The ScaledObject has been created successfully."))
+
+	return ctrl.Result{}, nil
+}
+
+func isEventScalerEnabled(workload client.Object) bool {
+	//TODO: define based on annotation
+	return true
+}
+
+func isRecoGenerated(conditions []metav1.Condition) bool {
+	for _, condition := range conditions {
+		if condition.Type == string(v1alpha1.RecoTaskProgress) {
+			if condition.Reason == RecoTaskRecommendationGenerated {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func isInitialized(conditions []metav1.Condition) bool {
+	for _, condition := range conditions {
+		if condition.Type == string(v1alpha1.Initialized) {
+			if condition.Status == metav1.ConditionTrue {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *HPAEnforcementController) SetupWithManager(mgr ctrl.Manager) error {
+	// when the max replicas change or when the deployment/rollout generation changes
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &kedaapi.ScaledObject{}, scaledObjectField, func(rawObj client.Object) []string {
```

Comment:
This IndexField for scaledObjectField is already in the main.go. Please check once if it is required here as well.